litespi/spi_nor_flash_module.py

```diff
@@ -0,0 +1,131 @@
+from litespi.spi_nor_config import SpiNorConfig
+from litespi.opcodes import SpiNorFlashOpCodes
+
+
+class SpiNorFlashModule:
+    """SPI NOR flash module
+
+    The ``SpiNorFlashModule`` class provides an essential information about chip capabilities
+    along with a supported read command.
+
+    Parameters
+    ----------
+    spi_nor_cfg : SpiNorConfig
+        User defined configuration for a chip.
+
+    Attributes
+    ----------
+    manufacturer_id : SpiNorFlashManufacturerIDs
+        ID of a vendor
+
+    device_id : int
+        ID of a chip
+
+    name : str
+        name of a chip
+
+    total_size : int
+        total size of a chip in bytes
+
+    page_size : int
+        page size of a chip in bytes
+
+    supported_modes : SpiNorConfig
+        supported access modes of a chip
+
+    addr_width : int
+        number of lines used to send address
+
+    addr_bits : int
+        number of address bits (24 or 32)
+
+    dummy_bits : int
+        number of dummy bits to send needed for a correct transfer
+
+    command : int
+        command used to read the memory
+
+    bus_width : int
+        expected number of data lines
+    """
+    def _check_xfer_mode(self, cfg):
```

Comment:
This changes `self` values, hence it should be called something like `_set_xfer_mode` or `_workout_best_xfer_mode` etc.

This function should have a docstring.

Reply:
I renamed it to [_configure_transfer_mode](https://github.com/litex-hub/litespi/pull/1/files#diff-0114c117482b2e8cd170c5a98f7a90d1R58) and I added a docstring.